file_store/src/file_info_poller.rs

```diff
@@ -171,8 +174,11 @@ where
         stream,
     };
 
-    let _ = sender.send(incoming_data_stream).await;
-    Ok(())
+    match sender.try_send(incoming_data_stream) {
+        Ok(_) => Ok(true),
+        Err(TrySendError::Full(_)) => Ok(false),
```

Comment:
It is not treating the channel being at capacity as an error. Returning false causes it to stop trying to send any more file across the channel and just wait for the next interval tick to try again.  I could return an Error here and than catch it in the calling function, but it felt simpler to just return a boolean rather than add another variant to the file_store/error.rs just to catch it in the caller and not bubble it up.